sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbird/integration/service/AuthenticationHelperService.java

```diff
@@ -0,0 +1,154 @@
+
+    @Autowired
+    private SignatureService signatureService;
+    @Autowired
+    private RestTemplate restTemplate;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+
+    protected static LocalDateTime getUTCDateTime() {
+        return ZonedDateTime
+                .now(ZoneOffset.UTC).toLocalDateTime();
+    }
+
+    public boolean isSupportedOtpChannel(String channel) {
+        return channel != null && otpChannels.contains(channel.toLowerCase());
+    }
+
+
+
+    public KycAuthResult doKycAuthImpl(String relyingPartyId, String clientId, KycAuthDto kycAuthDto)
+            throws KycAuthException {
+        try {
+            KycAuthRequestDto kycAuthRequestDto = new KycAuthRequestDto();
+            kycAuthRequestDto.setTransactionId(kycAuthDto.getTransactionId());
+            kycAuthRequestDto.setIndividualId(kycAuthDto.getIndividualId());
+
+            for (AuthChallenge authChallenge : kycAuthDto.getChallengeList()) {
+                if(Objects.equals(authChallenge.getAuthFactorType(),"KBA")){
+                    return knowledgeBasedAuthentcation(kycAuthDto.getIndividualId(),authChallenge);
+                }
+                throw new KycAuthException("invalid_challenge_format");
+            }
+        } catch (KycAuthException e) {
+            throw e;
+        } catch (Exception e) {
+            log.error("KYC-auth failed with transactionId : {} && clientId : {}", kycAuthDto.getTransactionId(),
+                    clientId, e);
+        }
+        throw new KycAuthException(ErrorConstants.AUTH_FAILED);
+    }
+
+    private KycAuthResult knowledgeBasedAuthentcation(String policyNumber, AuthChallenge authChallenge) throws KycAuthException {
+
+        KycAuthResult  kycAuthResult= new KycAuthResult();
+        SearchRequestDto searchRequestDto=new SearchRequestDto();
+        searchRequestDto.setLimit(2);
+        searchRequestDto.setOffset(0);
+        Map<String,Map<String,String>> filter=new HashMap<>();
+        Map<String,String> fullNameMap=new HashMap<>();
+        Map<String,String> dobMap=new HashMap<>();
+        Map<String,String> policyMap=new HashMap<>();
+        String encodedChallenge=authChallenge.getChallenge();
+
+        byte[] base64Bytes = encodedChallenge.getBytes(StandardCharsets.UTF_8);
+        byte[] decodedBytes = Base64.getDecoder().decode(base64Bytes);
+        String challenge = new String(decodedBytes, StandardCharsets.UTF_8);
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(challenge);
+
+            String fullName = jsonNode.get("name").asText();
+            String dob = jsonNode.get("dob").asText();
+
+
+            fullNameMap.put(fieldKey,fullName);
+            dobMap.put(fieldKey,dob);
+            policyMap.put(fieldKey,policyNumber);
+
+            filter.put(Full_Name,fullNameMap);
+            filter.put(DOB,dobMap);
+            filter.put(Policy_Number,policyMap);
```

Comment:
Above reading from challenge and construction of request should be dynamic based on the below config
mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-details

Since this will change based on the usecase and our plugin implementation should be generic

Reply:
dynamic request done